kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/processors/Watermarker.java

```diff
@@ -0,0 +1,207 @@
+
+        try {
+            srcFileImage = ImageIO.read(srcFile);
+        } catch (IOException ex) {
+            LOGGER.error("Could not open source file " + srcFile.getName() + ": " + ex.toString());
+            throw ex;
+        }
+
+        try {
+            LOGGER.info("Adding watermark to file, using " + conversionPropertiesWatermark.getRenderMode() + " mode");
+
+            final Integer defaultX = conversionPropertiesJpeg.getDefaultSize();
+            final Double scale = ((double) srcFileImage.getWidth() / defaultX) * 100;
+            IMOperation operation = new IMOperation();
+
+            if (conversionPropertiesCanvasExtension.isEnabled()) {
+                extendCanvas(srcFile, destFile, srcFileImage, scale);
+            }
+
+            switch (conversionPropertiesWatermark.getRenderMode()) {
+                case "text": {
+                    String drawCmd = "text "
+                            + getScaledValue(conversionPropertiesWatermarkTextMode.getOffsetX(), scale)
+                            + ","
+                            + getScaledValue(conversionPropertiesWatermarkTextMode.getOffsetY(), scale)
+                            + " \'"
+                            + conversionPropertiesWatermarkTextMode.getContent()
+                            + "\'";
+
+                    Integer fontSize = getScaledValue(conversionPropertiesWatermarkTextMode.getSize(), scale);
+
+                    /* Get the rgb color value from configuration */
+                    String[] fontColorArr = conversionPropertiesWatermarkTextMode.getColorRGB().split(",");
+                    Color fontColor = new Color(
+                            Integer.parseInt(fontColorArr[0].trim()),
+                            Integer.parseInt(fontColorArr[1].trim()),
+                            Integer.parseInt(fontColorArr[2].trim())
+                    );
+                    String fontColorRGB = "rgb(" + fontColor.getRed() + "," + fontColor.getGreen() + "," + fontColor.getBlue() + ")";
+
+                    operation.addImage(srcFile.getAbsolutePath());
+                    operation.colorspace("RGB");
+                    operation.font(conversionPropertiesWatermarkTextMode.getFont());
+                    operation.gravity(conversionPropertiesWatermarkTextMode.getGravity());
+                    operation.pointsize(fontSize);
+                    operation.fill(fontColorRGB);
+                    operation.draw(drawCmd);
+                    operation.addImage(destFile.getAbsolutePath());
+
+                    ImageCommand watermarkCmd = new ConvertCmd(conversionProperties.isUseGraphicsMagick());
+                    watermarkCmd.run(operation);
+                    break;
+                }
+                case "image": {
+                    String watermarkImageFilePath = conversionPropertiesWatermarkImageMode.getPath()
+                            + conversionPropertiesWatermarkImageMode.getFile();
```

Comment:
To concatenate paths like this is dangerous. It will crash if the path is /someting/without/slash/at/the/end. You can use new File(String path, String filename) instead. In this case it is however questionable if you really need a separate path, i see you use it only once. Why not simple filepath: /path/to/watermark.png?